src/librustc_target/spec/fuchsia_base.rs

```diff
@@ -24,9 +26,13 @@ pub fn opts() -> TargetOptions {
         dynamic_linking: true,
         executables: true,
         target_family: Some("unix".to_string()),
+        is_like_fuchsia: true,
         linker_is_gnu: true,
         has_rpath: false,
-        pre_link_args: args,
+        pre_link_args: pre_link_args,
+        pre_link_objects_exe: vec![
+            "Scrt1.o".to_string()
```

Comment:
This is causing `Scrt1.o` to be looked for inside `lib/rustlib/x86_64-fuchisa/lib` (the Rust sysroot) which is where liballoc, libcore, libcompiler_builtins, etc. live, not in the system sysroot where `Scrt1.o` lives.

Reply:
I think this is because of some windows support, see `src/rtstartup`.